src/plugins/visualizations/public/actions/add_agg_vis_action.ts

```diff
@@ -31,10 +33,16 @@ export class AddAggVisualizationPanelAction implements Action<EmbeddableApiConte
   public readonly type = ADD_AGG_VIS_ACTION_ID;
   public readonly id = ADD_AGG_VIS_ACTION_ID;
   public readonly grouping = [COMMON_EMBEDDABLE_GROUPING.legacy];
+  private readonly aggVisualizationCreationEnabled: boolean;
 
   public readonly order = 20;
 
-  constructor() {}
+  constructor(visTypes: TypesStart) {
+    this.aggVisualizationCreationEnabled =
+      visTypes.all().filter((type) => {
+        return !type.disableCreate && type.group === VisGroups.AGGBASED;
+      }).length > 0;
```

Comment:
rather than filter and compare, `some` would just exist when at least one matches the criteria.

```suggestion
      visTypes.all().some((type) => {
        return !type.disableCreate && type.group === VisGroups.AGGBASED;
      });
```